app/dataRoutes/__tests__/index.test.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+
+import Home from 'pages/Home';
+import Counter from 'pages/Counter';
+import NotFound from 'pages/NotFound';
+
+import dataRoutes from '../index';
+
+function MockedHome() {
+  return (
+    <h1>Home Page</h1>
+  );
+}
+
+function MockedCounter() {
+  return (
+    <h1>Counter Page</h1>
+  );
+}
+
+function MockedNotFound() {
+  return (
+    <h1>NotFound Page</h1>
+  );
+}
+
+jest.mock('pages/Home', function MockHome() {
+  return MockedHome;
+});
+jest.mock('pages/Counter', function MockCounter() {
+  return MockedCounter;
+});
+jest.mock('pages/NotFound', function MockNotFound() {
+  return MockedNotFound;
+});
+
+function findRouteByPath(routes, path) {
+  return routes.find(function matchRouteToPath(route) {
+    return route.path === path;
+  });
+}
+
+describe('dataRoutes', () => {
+  test('is an array of route objects', () => {
+    expect(dataRoutes).toEqual(expect.any(Array));
+
+    for (const route of dataRoutes) {
+      expect(route).toEqual(expect.objectContaining({
+        path: expect.any(String),
+      }));
+    }
+  });
+
+  describe('home route "/"', () => {
```

Comment:
Are we sure we want to test all routes individually?

Reply:
I think there is benefit to this:
* Ensures no mistakes are made in removal of intended routes as tests will also have to be updated on route changes
* It allows more granular tests in the future on [specific route configurations/behaviors](https://reactrouter.com/en/main/routers/picking-a-router#data-apis), especially around use of [loader data](https://reactrouter.com/en/main/hooks/use-loader-data), and conditional logic that results in redirects.